test/runtests.jl

```diff
@@ -634,6 +646,95 @@ end
             @test Box629{Int}(1) == Box629{Any}(1)
             @test hash(Box629{Int}(1)) == hash(Box629{Any}(1))
         end
+
+        @testset "ensure that type_seed(x) is stable" begin
+            @test 0x4965055ca9c88623 === type_seed(Int)
+            @test 0x0df062180f6b5880 === type_seed(String)
+
+            @test 0xf9a6ca5801d45b09 === type_seed(G)
+            @test 0xf9a6ca5801d45b09 === type_seed(G{T, U} where { T, U })
+            @test 0xf53284f9e807b4a3 === type_seed(G{T, U} where { T <: Int, U <: String })
+            @test 0x5cb3bb63536c6cf1 === type_seed(G{Int, String})
+            @test 0xa7672606502d460e === type_seed(G{Int, T} where T)
+            @test 0x355f38500af67860 === type_seed(G{T, String} where T)
+            @test 0x9f30a357e851217f === type_seed(G{T, T} where T)
+            @test 0x9f30a357e851217f === type_seed(G{T, T} where T)
+            @test 0x3d744a467126be1e === type_seed(G{G{T, Int}, G{T, String}} where T)
+            @test 0x3d744a467126be1e === type_seed(G{G{T, Int}, G{T, String}} where T)
+
+            @test 0x538beca01f293a83 === type_seed(Q)
+            @test 0x77cb5d6464d056b2 === type_seed(B)
+            @test 0x77cb5d6464d056b2 === type_seed(B{T} where { T })
+            @test 0x16d5d946443d9674 === type_seed(B{T} where { T <: Int })
+            @test 0x99cfdc2e3e9a8840 === type_seed(B{Int64})
+
+            @test 0xba0f522abd7271e6 === type_seed(Any)
+            @test 0x9514191d828c99aa === type_seed(Union{Int, String})
+            @test 0x3cef2865f9232667 === type_seed(Union{})
+            @test 0xb526450531768469 === type_seed(Union)
```

Comment:
ah - is this why you broke out `Union{}`?

Reply:
Yes.